joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForStructuresCreator.scala

```diff
@@ -0,0 +1,70 @@
+package io.joern.gosrc2cpg.astcreation
+
+import io.joern.x2cpg.ValidationMode
+import io.joern.gosrc2cpg.parser.ParserAst.*
+import io.joern.gosrc2cpg.parser.{ParserKeys, ParserNodeInfo}
+import io.joern.x2cpg
+import io.joern.x2cpg.{Ast, ValidationMode}
+import io.shiftleft.codepropertygraph.generated.{DispatchTypes, EdgeTypes, Operators}
+import ujson.Value
+import io.joern.x2cpg.datastructures.Stack.StackWrapper
+import io.shiftleft.codepropertygraph.generated.nodes.NewTypeDecl
+import scala.util.{Failure, Success, Try}
+
+import scala.util.Try
+
+trait AstForStructuresCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
+
+  def astForTypeSpec(typeSpecNode: ParserNodeInfo): Seq[Ast] = {
+    // TODO: Add support for member variables and methods
+    Option(typeSpecNode) match {
+      case Some(typeSpec) =>
+        val nameNode          = typeSpec.json(ParserKeys.Name)
+        val typeNode          = createParserNodeInfo(typeSpec.json(ParserKeys.Type))
+        val a: NewTypeDecl    = methodAstParentStack.collectFirst { case t: NewTypeDecl => t }.get
```

Comment:
Could we make this
```scala
methodAstParentStack.collectFirst { case t: NewTypeDecl => t } match
  Some(a) => ...
  None => Seq.empty
```

Reply:
Addressed here: https://github.com/joernio/joern/pull/3567/commits/7939c8bca109bad436a02b51d50e84e12403ce1d